src/docs/learn/architecture/test.md

```diff
@@ -0,0 +1,147 @@
+# Test Infrastructure
+
+::: info
+
+This article serves as an invitation for sharing ideas to improve our test infrastructure,
+feel free to contact me on [twitter](https://x.com/boshen_c) or [discord](https://discord.gg/9uXCAwqQZW).
+
+:::
+
+In Oxc, correctness and reliability are taken extremely seriously.
+
+We spend half of our time on strengthening the test infrastructure to prevent problems from propagating to downstream tools.
+
+## Parser
+
+### Memory Safety
+
+The oxc parser uses [`bumpalo`](https://docs.rs/bumpalo/latest/bumpalo) as the memory allocator for its AST.
+None of the AST nodes have a `drop` implementation.
+Miri [is used](https://github.com/oxc-project/oxc/actions/workflows/miri.yml) to ensure that no heap-allocated data is stored on the AST nodes
+so that memory is leaked when the memory allocator is dropped.
```

Comment:
Should we talk about memory safety in the lexer and SourceCode? Those use a lot of unsafe code.

Reply:
let's not mention `unsafe`.